### PR TITLE
Feature/paselc 546

### PR DESCRIPTION
**Pull request by GiacomoBrancazi**
#### List of Changes

add new api

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java [hunk 1]
```diff
@@ -26,31 +37,93 @@ public class UtilsController {
...
     @ApiOperation(value = "", notes = "${swagger.api.channels.getBrokerPsp}")
-    public BrokerOrPspDetailsResource getBrokerOrPspDetails(@ApiParam("swagger.request.brokerpspcode")
-                                                 @RequestParam(required = false, name = "brokerpspcode") String brokerPspCode) {
-        
-        
+    public BrokerOrPspDetailsResource getBrokerAndPspDetails(@ApiParam("swagger.request.brokerpspcode")
+                                                                 @PathVariable(required = true, name = "code") String brokerPspCode) throws Exception {
+        log.trace("getBrokerAndPspDetails start");
+        String xRequestId = UUID.randomUUID().toString();
+        log.debug("getBrokerAndPspDetails brokerPspCode = {} , xRequestId:  {}", brokerPspCode, xRequestId);
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java:55]:
```suggestion
```

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java [hunk 2]
```diff
@@ -26,31 +37,93 @@ public class UtilsController {
...
+    }
+
+    @GetMapping(value = "/ec-brokers/{code}/details", produces = {MediaType.APPLICATION_JSON_VALUE})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "", notes = "${swagger.api.channels.getBrokerPsp}")
+    public BrokerAndEcDetailsResource getBrokerAndEcDetails(@ApiParam("swagger.request.brokerEcCode")
+                                                            @PathVariable(required = true, name = "code") String brokerEcCode) throws Exception {
+        log.trace("getBrokerOrEcDetails start");
+        String xRequestId = UUID.randomUUID().toString();
+        log.debug("getBrokerOrEcDetails brokerPspCode = {} , xRequestId:  {}", brokerEcCode, xRequestId);
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java:96]:
```suggestion
   
```

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java [hunk 3]
```diff
@@ -26,31 +37,93 @@ public class UtilsController {
...
+        BrokersResource brokersResource = new BrokersResource();
+        brokersResource.setBrokers(new ArrayList<>());
+        CreditorInstitutionDetailsResource creditorInstitutionDetailsResource = null;
+        Brokers brokers;
+        CreditorInstitutionDetails creditorInstitutionDetails;
+
+        try {
+            brokers = apiConfigService.getBrokersEC(1, 0, brokerEcCode, null, null, "ASC");
+            brokersResource = BrokerMapper.toResource(brokers);
+        } catch (Exception e) {
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java:107]: use a specific exception (for example `FeignException.NotFound`)

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java [hunk 4]
```diff
@@ -26,31 +37,93 @@ public class UtilsController {
...
-        BrokerPspDetails brokerPspDetails = apiConfigService.getBrokerPsp(brokerPspCode);
-        BrokerPspDetailsResource brokerPspDetailsResource = ChannelMapper.toResource(brokerPspDetails);
+        BrokerPspDetailsResource brokerPspDetailsResource = null;
+        PaymentServiceProviderDetailsResource paymentServiceProviderDetailsResource = null;
+        BrokerPspDetails brokerPspDetails;
+        PaymentServiceProviderDetails paymentServiceProviderDetails;
+        try {
+            brokerPspDetails = apiConfigService.getBrokerPsp(brokerPspCode);
+            brokerPspDetailsResource = ChannelMapper.toResource(brokerPspDetails);
+        } catch (Exception e) {
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java:64]: ditto

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java [hunk 5]
```diff
@@ -26,31 +37,93 @@ public class UtilsController {
...
+            brokers = apiConfigService.getBrokersEC(1, 0, brokerEcCode, null, null, "ASC");
+            brokersResource = BrokerMapper.toResource(brokers);
+        } catch (Exception e) {
+            log.trace("getBrokerOrEcDetails - Not BrokerEC found");
+        }
+
+        try {
+            creditorInstitutionDetails = apiConfigService.getCreditorInstitutionDetails(brokerEcCode);
+            creditorInstitutionDetailsResource = mapper.toResource(creditorInstitutionDetails);
+        } catch (Exception e) {
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java:114]: same

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/stations/BrokerAndEcDetailsResource.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package it.pagopa.selfcare.pagopa.backoffice.web.model.stations;
+
+
+import it.pagopa.selfcare.pagopa.backoffice.web.model.creditorInstituions.CreditorInstitutionDetailsResource;
+import lombok.Data;
+
+@Data
+public class BrokerAndEcDetailsResource {
+
+    private BrokersResource brokerDetailsResource;
```
Comment (jacopocarlini) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/stations/BrokerAndEcDetailsResource.java:10]:
```suggestion
    private BrokerResource brokerDetailsResource;
```

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java (hunk 1 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java]: fixed

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java (hunk 3 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java]: fixed

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java (hunk 5 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java]: fixed

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/stations/BrokerAndEcDetailsResource.java (hunk 6 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/stations/BrokerAndEcDetailsResource.java]: fixed

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java (hunk 2 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java]: fixed

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java (hunk 4 above)
Comment (GiacomoBrancazi) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java]: fixed

**Review by jacopocarlini — APPROVED** [review]
Good Job! 💪

**Pull request merged**